### DISCUSSION TITLE
Search results

### DISCUSSION THREAD

**Discussion by larsinka** [mjs/imapclient#469] [General]
I'm trying to fetch email containing a certain keyword from my INBOX folder in my .iCloud account. Now the problem is that I only get some emails back, not all that should show up (e.g. If I search for the same keyword in my mail application I get 50 results, whereas my python script only gets 8). Im using the code from here: https://imapclient.readthedocs.io/en/2.2.0/

**Comment by mjs** [comment]
It's possible (likely in fact) that your email application caches messages locally and searches those rather than relying on an search abilities of the IMAP server.

Can you share the exact `search` call you were using? Something like `fetch(['BODY', "your search"])` or `fetch(['TEXT', "your search"])` might work.

Some IMAP servers (e.g. Gmail) have non-standard support for server side searches. What does `capabilities()` return when IMAPClient is connected to iCloud?

**Reply by larsinka** [reply]
Thanks, I got it to work eventually. I don't really remember what I did to solve it. 

The `search` call looks like this: `messages = client.search(['SUBJECT', 'mySubject'])`.

`client.capabilities()` returns `(b'XAPPLEPUSHSERVICE', b'IMAP4', b'IMAP4REV1', b'CONDSTORE', b'ENABLE', b'QRESYNC', b'QUOTA', b'XAPPLELITERAL', b'NAMESPACE', b'UIDPLUS', b'CHILDREN', b'BINARY', b'UNSELECT', b'SORT', b'CATENATE', b'URLAUTH', b'LANGUAGE', b'ESEARCH', b'ESORT', b'THREAD=ORDEREDSUBJECT', b'THREAD=REFERENCES', b'CONTEXT=SEARCH', b'CONTEXT=SORT', b'WITHIN', b'SASL-IR', b'SEARCHRES', b'METADATA', b'ID', b'XMSEARCH', b'X-SUN-SORT', b'ANNOTATE-EXPERIMENT-1', b'X-UNAUTHENTICATE', b'X-SUN-IMAP', b'XUM1', b'MULTISEARCH', b'IDLE', b'X-APPLE-REMOTE-LINKS', b'LIST-STATUS')`

Thank you for helping!

**Comment by mjs** [comment]
Great that you got it working.

It looks like you're only searching the message subjects. If you want to search the whole message your could trying swapping `SUBJECT` for `BODY` or `TEXT`.
